Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -2140,33 +2149,38 @@ - (void)collectionView:(JSQMessagesCollectionView *)collectionView
                 if ([[messageItem media] isKindOfClass:[TSPhotoAdapter class]]) {
                     TSPhotoAdapter *messageMedia = (TSPhotoAdapter *)[messageItem media];
 
-                    UIImage *tappedImage = ((UIImageView *)[messageMedia mediaView]).image;
+                    UIView *mediaView = [messageMedia mediaView];
+                    if (![mediaView isKindOfClass:[UIImageView class]]) {
+                        OWSFail(@"unexpected mediaView of type: %@", [mediaView class]);
+                        return;
+                    }
+                    UIImageView *imageView = (UIImageView *)mediaView;
+                    UIImage *tappedImage = imageView.image;
                     if (tappedImage == nil) {
                         DDLogWarn(@"tapped TSPhotoAdapter with nil image");
-                    } else {
-                        UIWindow *window = [UIApplication sharedApplication].keyWindow;
-                        JSQMessagesCollectionViewCell *cell
-                            = (JSQMessagesCollectionViewCell *)[collectionView cellForItemAtIndexPath:indexPath];
-                        OWSAssert([cell isKindOfClass:[JSQMessagesCollectionViewCell class]]);
-                        CGRect convertedRect = [cell.mediaView convertRect:cell.mediaView.bounds toView:window];
+                        return;
+                    }
```

Comment:
The rest of this block is just whitespace changes due to using an early return instead of if/else.